src/api/route-services/gem2s.js

```diff
@@ -49,7 +49,12 @@ class Gem2sService {
   static async generateGem2sTaskParams(experimentId) {
     const experiment = await (new ExperimentService()).getExperimentData(experimentId);
     const { samples } = await (new SamplesService()).getSamplesByExperimentId(experimentId);
-    const { metadataKeys } = await (new ProjectService()).getProject(experiment.projectId);
+    const {
```

Comment:
it re runs gem23s if we re order the samples and then click launch analysis, I think we could avoid this relatively easily.

In the api we would need to change the way we create the hash to check if we should re run or not the pipeline, one option could be by ordering the samples before generating the hash. (just ordering for the hash so that it isn’t order-sensitive, but not ordering the actual samples array that is going to be stored)

Reply:
Yup, this would fix the issue. The first time the project is launched after this update, it would re run GEM2S. However, on subsequent runs, it wouldn't.

Reply:
It seems that the problem is bigger than this. Suggests moving this to another ticket.